resources/internal_concordances.go

```diff
@@ -15,52 +19,38 @@ type internalConcordancesResponse struct {
 }
 
 // InternalConcordances concords provided uuids, and enriches them with concept model
-func InternalConcordances(concordances concepts.Concordances, search concepts.Search) func(w http.ResponseWriter, r *http.Request) {
+func InternalConcordances(concordances concepts.Concordances, search concepts.Search, publicConcepts concepts.PublicConcepts) func(w http.ResponseWriter, r *http.Request) {
 	return func(w http.ResponseWriter, req *http.Request) {
 		w.Header().Add("Content-Type", "application/json")
 		tid := tidutils.GetTransactionIDFromRequest(req)
 
-		authority := concepts.NoAuthority
-		authorityParam, foundAuthority := getMultiValuedParam(req, "authority")
-		if foundAuthority {
-			if len(authorityParam) != 1 {
-				writeJSON("Please provide one value for 'authority' query parameter", http.StatusBadRequest, w)
-				return
-			}
-			authority = authorityParam[0]
-			if authority == "" {
-				writeJSON("Please provide a non-empty 'authority' query parameter", http.StatusBadRequest, w)
-				return
-			}
+		authority, err := getAuthority(req)
+		if err != nil {
+			writeJSON(err.Error(), http.StatusBadRequest, w)
+			return
 		}
-		ids, idsFound := getMultiValuedParam(req, "ids")
-		if !idsFound {
-			writeJSON("Please provide ids to concord, using the 'ids' query parameter", http.StatusBadRequest, w)
+
+		ids, err := getIDs(req)
+		if err != nil {
+			writeJSON(err.Error(), http.StatusBadRequest, w)
 			return
 		}
 
-		includeDeprecated := true
-		includeDeprecatedParam, foundIncludeDeprecated := getMultiValuedParam(req, "include_deprecated")
-		if foundIncludeDeprecated {
-			if len(includeDeprecatedParam) != 1 {
-				writeJSON("Please provide one value for 'include_deprecated' query parameter", http.StatusBadRequest, w)
-				return
-			}
-			includeDeprecatedValue, err := strconv.ParseBool(includeDeprecatedParam[0])
-			if err != nil {
-				writeJSON("Please provide a valid boolean for 'include_deprecated' query parameter", http.StatusBadRequest, w)
-				return
-			}
-			includeDeprecated = includeDeprecatedValue
+		includeDeprecated, err := getIncludeDeprecated(req)
+		if err != nil {
+			writeJSON(err.Error(), http.StatusBadRequest, w)
+			return
 		}
-		identifiers, err := concordances.GetConcordances(tid, authority, ids...)
-		if err == concepts.ErrConceptIDsAreEmpty {
-			writeJSON("Please provide non-empty ids to concord, using the 'ids' query parameter", http.StatusBadRequest, w)
+
+		identifiers, err := getConcordances(concordances, tid, authority, ids)
+
+		if err == util.ErrConceptIDsAreEmpty {
+			writeJSON("please provide non-empty ids to concord, using the 'ids' query parameter", http.StatusBadRequest, w)
```

Comment:
`question`: Aren't we reading the concordances not creating them ?
```suggestion
			writeJSON("please provide non-empty ids to read, using the 'ids' query parameter", http.StatusBadRequest, w)
```

Reply:
I didn't change the message, I just moved the code and changed the first letter to lower.
This is according to the readme:
_Internal UPP Concordances API which accepts a list of concept IDs, concords them, and returns them as an array of UPP concepts._

Reply:
Ah, okay, the definition is a little bit weird, because in our platform concording concepts is a thing of itself as you know. This one is more like aggregating results in a json response like so 
```
{
    "concepts": {
        "019b431f-bfc3-3913-8e54-f0d3b423da50": {
            "id": "http://www.ft.com/thing/019b431f-bfc3-3913-8e54-f0d3b423da50",
            "apiUrl": "http://api.ft.com/people/019b431f-bfc3-3913-8e54-f0d3b423da50",
            "type": "http://www.ft.com/ontology/person/Person",
            "prefLabel": "Mary-Kate and Ashley Olsen",
            "isFTAuthor": false,
            "isDeprecated": true
        },
        "f4dfcb62-db9a-402e-b4b5-62bd60ad2038": {
            "id": "http://www.ft.com/thing/f4dfcb62-db9a-402e-b4b5-62bd60ad2038",
            "apiUrl": "http://api-t.ft.com/organisations/f4dfcb62-db9a-402e-b4b5-62bd60ad2038",
            "type": "http://www.ft.com/ontology/organisation/Organisation",
            "prefLabel": "Brewin Dolphin Holdings PLC"
        }
    }
}
``` rather than `concording` concepts in the KG. And basically this is why I was confused.

Reply:
I decided to leave it as it is, it will stay consistent with itself.

Reply:
Yeah, that is okay, I got your point!